chain/messagepool/messagepool.go

```diff
@@ -852,14 +875,14 @@ func (mp *MessagePool) addTs(ctx context.Context, m *types.SignedMessage, curTs
 		return false, xerrors.Errorf("minimum expected nonce is %d: %w", snonce, ErrNonceTooLow)
 	}
 
-	mp.lk.Lock()
-	defer mp.lk.Unlock()
-
 	senderAct, err := mp.api.GetActorAfter(m.Message.From, curTs)
 	if err != nil {
 		return false, xerrors.Errorf("failed to get sender actor: %w", err)
 	}
 
+	mp.lk.Lock()
```

Comment:
I think this can go a little further down (before the call to `verifyMsgBeforeAdd`, at minimum?)